### PR TITLE
Fixed a number of compiler warnings

### PR DESCRIPTION
**Pull request by simonduq**
This PR simply fixes some compiler warnings.

### CODE REVIEW

File: core/net/mac/contikimac/contikimac.c [hunk 1]
```diff
@@ -672,11 +664,11 @@ send_packet(mac_callback_t mac_callback, void *mac_callback_ptr,
...
     {
       rtimer_clock_t wt;
-      rtimer_clock_t txtime;
-      int ret;
-
-      txtime = RTIMER_NOW();
-      ret = NETSTACK_RADIO.transmit(transmit_len);
+      rtimer_clock_t txtime = RTIMER_NOW();
+#if RDC_CONF_HARDWARE_ACK
+      int ret =
```
Comment (adamdunkels) [inline core/net/mac/contikimac/contikimac.c]: Ouch, this one was extremely ugly! Should probably be rewritten :)

**Comment by adamdunkels** [comment]
Very good - one comment though.

File: core/net/mac/contikimac/contikimac.c (hunk 1 above)
Comment (simonduq) [inline core/net/mac/contikimac/contikimac.c]: agreed, will do

**Comment by g-oikonomou** [comment]
@simonduq have all of Adam's comments been taken care of?

**Comment by simonduq** [comment]
Opps, closed the PR by mistake. Now re-opened and Travis started again... (note I did not re-push or anything)
Yes, all comments from @adamdunkels and @cmorty have been addressed.

File: core/lib/petsciiconv.c [hunk 2]
```diff
@@ -73,13 +73,13 @@ static unsigned char ascii2petscii[128] = {
 };
 
 static unsigned int i;
-static unsigned char *ptr;
+static char *ptr;
```
Comment (cmorty) [inline core/lib/petsciiconv.c]: Oh, and looking at this file again. i and ptr do not need to be static variable*, but can be put into a local scope as automatic variables. Saves a few bytes of global data memory.

* Do not confuse with the static key-word. :)

Comment (simonduq) [inline core/lib/petsciiconv.c]: good point, will do!

**Comment by g-oikonomou** [comment]
:+1:

**Pull request merged**